pkg/bundle/internal/target/target.go

```diff
@@ -396,3 +417,48 @@ func (b *Data) Populate(pool *util.CertPool, formats *trustapi.AdditionalFormats
 	}
 	return nil
 }
+
+// Calculate hash from the given password string.
+func calculateHash(passwd string) string {
+	crcTable := crc32.MakeTable(128)
+	crc := crc32.New(crcTable)
+	crc.Write([]byte(passwd))
+
+	return strconv.Itoa(int(crc.Sum32()))
```

Comment:
I guess it is excessively to use sha256 for short text patterns like target password.
OK, I will rewrite.

Reply:
That's a fair point. I will consider the alternatives again. 😊

Reply:
So, do keep it as it is?

Reply:
I still think we should use a proper hash algorithm. And to keep it simple and consistent I think SHA 256 might be an ok choice. My main concern when reading about CRC (which is not well known to me):

> Secondly, unlike cryptographic hash functions, CRC is an easily reversible function, which makes it unsuitable for use in digital signatures.

While these truststore passwords are not very sensible secrets, I am pretty sure some users will think we should treat them well - as passwords. I understand your concern about size, but encoded as hex it is not that bad. To avoid calculating the hash for each target, I suggest building the hash in the main Bundle reconcile function. It can be supplied together with the existing data hash.

Maybe @inteon or @SgtCoDFish wants to chime in here?

Reply:
Exactly! As passwords are stored in insecure fields, no reason to produce secure hash.
One potential issue here is hash collision, when user changed password, but nothing happened because new calculated hash is the same one as for old password.

Reply:
We already use a sha256 hash here: https://github.com/cert-manager/trust-manager/blob/c32b33f6229eececd19a1457df483da34980a1cb/pkg/bundle/internal/target/target.go#L111

I would try and use the same mechanism (even combine them if possible).

Reply:
Got it. 
I'm gonna rewrite it to use sha256 hash function.